alibi_detect/saving/saving.py

```diff
@@ -110,7 +117,9 @@ def _cleanup_filepath(orig_files: set, filepath: Path):
 
 
 # TODO - eventually this will become save_detector (once outlier and adversarial updated to save via config.toml)
-def _save_detector_config(detector: ConfigurableDetector, filepath: Union[str, os.PathLike]):
+def _save_detector_config(detector: ConfigurableDetector,
+                          filepath: Union[str, os.PathLike],
+                          save_state: bool = False):
```

Comment:
Should `save_state` default to False for stateful detectors? In my mind, I'd assume when saving it would save everything by default. But perhaps that's not the intended use case?

Reply:
I notice also that the default behaviour for `load_detector` is to load the detector's state if it exists in the save directory. I guess if we have the above saving functionality perhaps we should have symmetric load functionality. So: `load_detector(dir, load_state=False)`.

Reply:
I think in general the best thing is to have the defaults be the typical user expectations. So save would default to saving everything in the same way load defaults to loading everything but have the extra functionality that allows them to ignore state while saving if they choose. I'm not sure thought, but perhaps there's a good reason to expect that save not include state...

Reply:
> I notice also that the default behaviour for `load_detector` is to load the detector's state if it exists in the save directory. I guess if we have the above saving functionality perhaps we should have symmetric load functionality. So: `load_detector(dir, load_state=False)`.

I'm not so sure about complicating `load_detector` with addition options. It seems quite natural to me that `load_detector` will load whatever you saved in `filepath` with `save_detector`. In other words, deciding what to save is done only at save time. The user could always through away the state later by running `dd = load_detector()` and then `dd.reset()`.

Reply:
> I think in general the best thing is to have the defaults be the typical user expectations. So save would default to saving everything in the same way load defaults to loading everything but have the extra functionality that allows them to ignore state while saving if they choose. I'm not sure thought, but perhaps there's a good reason to expect that save not include state...

I am leaning toward agreeing with you wrt to by default saving online state. Offline state will need more thought in the future I think. Saving offline state might limit the degree to which "config" can be later edited (e.g. offline state would include saving `self.thresholds`, which would prevent the `window_size` being changed at a later date). In other words, IMO adopting a rule that we save/and load EVERYTHING might be a step too far. Referring back to my last comment, I think its more about saving "whatever makes sense" to `filepath`, and then `load_detector` loads EVERYTHING that is in `filepath`.

Reply:
This is pretty important to decide on (ofc we can change default behaviour later), what is the more common use case after developing an online detector and deciding to deploy it - not sure the answer is so clear? @ojcobb @arnaudvl 

Regardless of behaviour, this needs to be explicit in the docs (not just docstrings imo).